src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/EncryptedReference.cs

```diff
@@ -1,16 +1,17 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Xml;
 
 namespace System.Security.Cryptography.Xml
 {
     public abstract class EncryptedReference
     {
-        private string _uri;
-        private string _referenceType;
-        private TransformChain _transformChain;
-        internal XmlElement _cachedXml;
+        private string _uri = null!;
```

Comment:
Mark `Uri` property's setter with `[MemberNotNull(nameof(_uri))]`, and you won't need to do ` = null!`

Reply:
Thank you @maxkoshevoi !